Lesson 03 Submitted for Review

srepking opened the pull request.

charlesrobison commented on students/srepking/Lesson03/Slicing.py:
```diff
@@ -0,0 +1,59 @@
+            print("Your list is not divisible by three the second time.")
+    else:
+        print('This list is not divisible by three')
+    return
```
Is this return statement complete?  Assignment was to return the first third, then last third, then middle third.  So, 1,2,3,4,5,6,7,8,9 would return 7,8,9,1,2,3,4,5,6.  Can be simplified to:
`i = len(x) // 3`
`return x[-1:] + x[:i] + x[i:-i]`

charlesrobison commented on students/srepking/Lesson03/Slicing.py:
```diff
@@ -0,0 +1,59 @@
+
+print('Creates a list with the middle third and last third, then the first third in the new order')
+print(thirds([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 2, 13, 14, 15, 16, 17, 18]))
+
```
All of the functions for this assignment should have tests.

charlesrobison commented on students/srepking/Lesson03/list_lab.py:
```diff
@@ -0,0 +1,84 @@
+# Display all the fruits that begin with P, using a for loop
+new_list = []
+for i in fruits:
+     if i[0] == 'P':
```
Recommend adding 'lower' to catch case issues.  e.g., `if i[0].lower() = 'p':`

charlesrobison commented on students/srepking/Lesson03/mailroom.py:
```diff
@@ -0,0 +1,86 @@
+            report()
+        if response == '3':
+            print('Goodbye!')
+            break
```
Works well!

charlesrobison commented on students/srepking/Lesson03/strformat_lab.py:
```diff
@@ -0,0 +1,57 @@
+#Start of Task 5
+print('\n''This is the beginning of Task 5')
+
+ourlist = ['orange', 1.3, 'lemon', 1.1]
```
Per PEP8 variable names should be lowercase and words should be separated by underscores: `our_list`

charlesrobison commented on students/srepking/Lesson03/strformat_lab.py:
```diff
@@ -0,0 +1,57 @@
+# Start of Task 6
+print('\n''This is the beginning of Task 6')
+name = ('Buick', 'Honda', 'Porcheeeeeeee')
+wiN = int(len(str(max(name))))+2
```
This should be PEP8, lower case and separated by underscores, and something more descriptive.  Same for wiA below.

natasha-aleksandrova commented on students/srepking/Lesson03/mailroom.py:
```diff
@@ -0,0 +1,86 @@
+            thank_you()
+        if response == '2':
+            report()
+        if response == '3':
```
this should be an if/elif not if/if since it can only be one of those things.

srepking commented:
Thank you for the feedback! I am going to pull the changes to git hub now.

On Sun, Aug 26, 2018 at 5:34 PM Charles Robison <notifications@github.com>
wrote:

> *@charlesrobison* approved this pull request.
>
> Code works well. Few recommendations. Be sure to add tests when the
> assignments require them. This is ready to merge.
> ------------------------------
>
> In students/srepking/Lesson03/Slicing.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212843864>
> :
>
> > +def thirds(x):
> +    """Creates a list with the middle third and last third, then the first
> +    third in the new order"""
> +    if len(x) % 3 == 0:
> +        # if list is divisible by three, make a new list out of the last two thirds.
> +        third = int(len(x)/3)
> +        new_list = (x[-(2*third):])
> +        # If the new list is divisible by three, take the first third in the new list.
> +        if len(new_list) % 3 ==0:
> +            new_third = int(len(new_list)/3)
> +            return new_list[0:new_third]
> +        else:
> +            print("Your list is not divisible by three the second time.")
> +    else:
> +        print('This list is not divisible by three')
> +    return
>
> Is this return statement complete? Assignment was to return the first
> third, then last third, then middle third. So, 1,2,3,4,5,6,7,8,9 would
> return 7,8,9,1,2,3,4,5,6. Can be simplified to:
> i = len(x) // 3
> return x[-1:] + x[:i] + x[i:-i]
> ------------------------------
>
> In students/srepking/Lesson03/Slicing.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212843888>
> :
>
> > +        third = int(len(x)/3)
> +        new_list = (x[-(2*third):])
> +        # If the new list is divisible by three, take the first third in the new list.
> +        if len(new_list) % 3 ==0:
> +            new_third = int(len(new_list)/3)
> +            return new_list[0:new_third]
> +        else:
> +            print("Your list is not divisible by three the second time.")
> +    else:
> +        print('This list is not divisible by three')
> +    return
> +
> +
> +print('Creates a list with the middle third and last third, then the first third in the new order')
> +print(thirds([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 2, 13, 14, 15, 16, 17, 18]))
> +
>
> All of the functions for this assignment should have tests.
> ------------------------------
>
> In students/srepking/Lesson03/list_lab.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212844001>
> :
>
> > +print('\n'"Nice Choice! You chose {number}, which corresponds to the {fruit}"'\n'.format(
> +number=response, fruit=fruits[int(response)-1]))
> +
> +# Add another to the beginning using '+'
> +fruits=['tomato']+fruits
> +print('\n''We added tomatoes to the beginning of the list')
> +print(fruits)
> +# Add another to the beginning using 'insert'
> +fruits.insert(0, 'kiwi')
> +print('\n''We added kiwis to the beginning of the list')
> +print (fruits)
> +
> +# Display all the fruits that begin with P, using a for loop
> +new_list = []
> +for i in fruits:
> +     if i[0] == 'P':
>
> Recommend adding 'lower' to catch case issues. e.g., if i[0].lower() =
> 'p':
> ------------------------------
>
> In students/srepking/Lesson03/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212844470>
> :
>
> > +            print(f"{key:{name_wi}}${temp[0]:{tot_wi}}{temp[1]:^{num_wi}}   ${temp[2]:>{ave_wi}.2f}")
> +
> +        print('\n')
> +
> +
> +if __name__ == '__main__':
> +    donors= {'Joe Edgar Allen Poe the Third ':[1,4,200],'Jack':[4,5], 'Jill':[4], 'Jake':[.30],'Jim':[1,2,1.04]}
> +    while True:
> +        response = input('Type \'1\' to Send a Thank You, \'2\' to Create a Report, or \'3\' to quit?')
> +        if response == '1':
> +            thank_you()
> +        if response == '2':
> +            report()
> +        if response == '3':
> +            print('Goodbye!')
> +            break
>
> Works well!
> ------------------------------
>
> In students/srepking/Lesson03/strformat_lab.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212845163>
> :
>
> > +    return f"{first}{second}"
> +
> +print(formatter((2,3,4,5,6,7,8)))
> +
> +
> +#Start of Task 4
> +print('\n''This is the beginning of Task 4')
> +y = (4,30,2017,2,27)
> +
> +print("{} {} {} {} {}".format((str(y[3])).zfill(2),
> +str(y[4]).zfill(2), str(y[2]).zfill(2), str(y[0]).zfill(2), str(y[1])).zfill(2))
> +
> +#Start of Task 5
> +print('\n''This is the beginning of Task 5')
> +
> +ourlist = ['orange', 1.3, 'lemon', 1.1]
>
> Per PEP8 variable names should be lowercase and words should be separated
> by underscores: our_list
> ------------------------------
>
> In students/srepking/Lesson03/strformat_lab.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#discussion_r212845210>
> :
>
> > +
> +#Start of Task 5
> +print('\n''This is the beginning of Task 5')
> +
> +ourlist = ['orange', 1.3, 'lemon', 1.1]
> +print(f"The weight of an {ourlist[0]} is {ourlist[1]} \
> +and the weight of a {ourlist[2]} is {ourlist[3]}")
> +
> +print('\n''Fruits in upper case and the weight 20% higher')
> +print(f"The weight of an {ourlist[0].upper()} is {ourlist[1]*1.2} \
> +and the weight of a {ourlist[2].upper()} is {ourlist[3]*1.2}")
> +
> +# Start of Task 6
> +print('\n''This is the beginning of Task 6')
> +name = ('Buick', 'Honda', 'Porcheeeeeeee')
> +wiN = int(len(str(max(name))))+2
>
> This should be PEP8, lower case and separated by underscores, and
> something more descriptive. Same for wiA below.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/923#pullrequestreview-149555660>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AobQtRiaKF2-9igkR1zHP-iYDbvYM0Twks5uUz6PgaJpZM4WM5Sn>
> .
>
